homeassistant/components/gardena/__init__.py

```diff
@@ -0,0 +1,78 @@
+"""
+Support for Gardena Smart sytem devices.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/gardena/
+"""
+import logging
+import voluptuous as vol
+from gardena.smart_system import SmartSystem
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import async_load_platform
+from homeassistant.const import CONF_PASSWORD, CONF_EMAIL
+
+REQUIREMENTS = ["py-smart-gardena==0.6.15"]
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "gardena"
+GARDENA_SYSTEM = "gardena_system"
+GARDENA_LOCATION = "gardena_location"
+GARDENA_CONFIG = "gardena_config"
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_EMAIL): cv.string,
+                vol.Required(CONF_PASSWORD): cv.string,
+                vol.Required("client_id"): cv.string,
+                vol.Required("location_id"): cv.string,
+                vol.Optional("default_mower_duration_in_minutes"): cv.string,
+                vol.Optional(
+                    "default_smart_irrigation_control_duration_in_minutes"
+                ): cv.string,
+                vol.Optional("default_smart_watering_duration_in_minutes"): cv.string,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, config):
+    """Set up the gardena component."""
+    _LOGGER.debug("Initialising Gardena")
+
+    hass.data[GARDENA_SYSTEM] = GardenaSmartSystem(hass, config[DOMAIN], SmartSystem)
+    hass.data[GARDENA_CONFIG] = config[DOMAIN]
+    _LOGGER.debug("Gardena component initialised")
+    for component in ("vacuum", "sensor", "switch"):
+        hass.async_create_task(async_load_platform(hass, component, DOMAIN, {}, config))
+
+    return True
+
+
+class GardenaSmartSystem:
+    """A Gardena Smart System wrapper class."""
+
+    def __init__(self, hass, domain_config, smart_system):
+        """Initialize the Gardena Smart System."""
+        self.config = domain_config
+        self._hass = hass
+
+        self.smart_system = smart_system(
+            domain_config[CONF_EMAIL],
+            domain_config[CONF_PASSWORD],
+            domain_config["client_id"],
```

Comment:
```suggestion
            domain_config[CONF_CLIENT_ID],
```

with `"CONF_CLIENT_ID = client_id"` additionally.